modules/basicfuncs/tf-tag.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023 Balazs Scheidler <balazs.scheidler@axoflow.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+typedef struct _TFTagState
+{
+  LogTagId tag_id;
+  GString *value_if_set;
+  GString *value_if_unset;
```

Comment:
Don't we want to let the user use templates here?

Reply:
I am not sure. If there's a usecase then both the tag name or the expanded values could become templates. But for now my intended use-case has been to convert a tag to a label in metrics-probe.

The value-if-set/unset are only there to make it easy to use a more descriptive value than 0 or 1, which this normally expands to.

So with this, I'd wait until a use-case emerges.

Reply:
It would be useful for AND-ing and OR-ing different tags.

Reply:
We can leave it as is for now, and extend it later, sure.